src/classes/CRLP_ApiService.cls

```diff
@@ -194,8 +213,329 @@ public without sharing class CRLP_ApiService {
+    private RollupMetadataHandler getRollupHandler(Set<String> changedRollupTypes) {
+        RollupMetadataHandler callback = new RollupMetadataHandler();
+
+         if (!changedRollupTypes.isEmpty()) {
+            callback.params = new Map<String, Set<String>> {
+                CallableApiParameters.PARAM_ROLLUP_TYPES => changedRollupTypes
+            };
+        }
+
+        return callback;
+    }
+
+    /*
+    * @description build a set of changed rollup types based on a list of rollups
+    * @param rollups is a list of rollups
+    * @return a set of change rollup types
+    */
+
+    private Set<String> buildChangedRollupTypes(List<CRLP_RollupCMT.Rollup> rollups) {
+        Set<String> changedRollupTypes = new Set<String>();
+
+        if (!isRollupStateEnabled()) {
+            return changedRollupTypes;
+        }
+
+        for (CRLP_RollupCMT.Rollup cmtRollup : rollups) {
+            RollupUtil util = new RollupUtil(cmtRollup);
+            if (util.hasChanged()) {
+                changedRollupTypes.addAll(util.getRollupTypes());
+            }
+        }
+
+        return changedRollupTypes;
+    }
+
+    /*
+    * @description build a set of changed rollup types based on a list of filter groups
+    * @param filterGroups is a List of filtergroups
+    * @return a set of change rollup types
+    */
+    private Set<String> buildChangedRollupTypes(List<CRLP_RollupCMT.FilterGroup> filterGroups) {
+        Set<String> changedRollupTypes = new Set<String>();
+
+        if (!isRollupStateEnabled()) {
+            return changedRollupTypes;
+        }
+
+        FilterGroupUtil util = new FilterGroupUtil();
+        for (CRLP_RollupCMT.FilterGroup cmtFilterGroup : filterGroups) {
+            if (util.hasChanged(cmtFilterGroup)) {
+                changedRollupTypes.addAll(util.getRollupTypes(cmtFilterGroup.recordId));
+            }
+        }
+
+        return changedRollupTypes;
+    }
+
+    /*
+    * @description handle changes to customizable rollups and send back a callback handler if 
+    * there is processing that needs to be executed after deloyment
+    * @param rollups is a List of Rollups
+    * @return a MetadataHandler that can be used for deployment callbacks
+    */
+    public RollupMetadataHandler sendChangeEvent(List<CRLP_RollupCMT.Rollup> rollups) {
+
+        Set<String> changedRollupTypes = new Set<String>();
+        changedRollupTypes.addAll(buildChangedRollupTypes(rollups));
+
+       return getRollupHandler(changedRollupTypes);
+    }
+
+    /*
+    * @description handle changes to customizable filter groups and send back a callback handler if
+    * there is processing that needs to be executed after deployment
+    * @param filterGroups is a list of filtergroups
+    * @return a MetadataHandler that can be used for deployment callbacks
+    */
+    public RollupMetadataHandler sendChangeEvent(List<CRLP_RollupCMT.FilterGroup> filterGroups) {
+        Set<String> changedRollupTypes = new Set<String>();
```

Comment:
These two lines can be rewritten as:
```
Set<String> changedRollupTypes = buildChangedRollupTypes(filterGroups);
```

Reply:
Thanks - yes I have combined them.  Good catch.